pkg/kube/sourcerepositories.go

```diff
@@ -39,13 +40,42 @@ func GetRepositoryGitURL(s *v1.SourceRepository) (string, error) {
 	return spec.HTTPCloneURL, nil
 }
 
-// FindSourceRepository returns a SourceRepository for the given namespace, owner and name
-// or returns an error if it cannot be found
-func FindSourceRepository(jxClient versioned.Interface, ns string, owner string, name string) (*v1.SourceRepository, error) {
+// FindSourceRepositoryWithoutProvider returns a SourceRepository for the given namespace, owner and repo name.
+// If no SourceRepository is found, return nil.
+func FindSourceRepositoryWithoutProvider(jxClient versioned.Interface, ns string, owner string, name string) (*v1.SourceRepository, error) {
+	return FindSourceRepository(jxClient, ns, owner, name, "")
+}
+
+// FindSourceRepositoryByLabels returns a SourceRepository matching the given label selector, if it exists.
+func FindSourceRepositoryByLabels(jxClient versioned.Interface, ns string, labelSelector string) (*v1.SourceRepository, error) {
```

Comment:
I find it funny that in one case we have three pieces of information to identify a repo and in the other two. That seems odd to me. If the provider is needed, why is it not needed for the lookup by name? I guess once we would have multiple providers and we allow for using multiple providers in parallel, we can get ambiguity with `FindSourceRepositoryWithoutProvider`. 

My guess is that we don't add the provider to the CRD name, because that would make it even easier to hist the name length limit. Maybe we should get rid of `FindSourceRepositoryWithoutProvider` and place our bet on label lookup only?

Regarding, _labelSelector_, why putting the burden on how to build the selector on the caller? How does the caller know the right label format? Would you not pass _name_, _owner_ and _provider_ and let _FindSourceRepositoryByLabels_ sort this out?

Reply:
Basically, this is because `jx start pipeline` doesn’t have the provider URL available yet when it calls this stuff - in fact, it uses the `SourceRepository` to find the provider URL! I’m probably overdoing things here with the provider to hedge for future issues, if/when we properly support multiple providers on a single cluster, but the more I think about it, the more I think I’m probably making it more complicated than it needs to be. I’ll remove the provider from the label and change the function to take org and repo as its arguments.